cylc/flow/config.py

```diff
@@ -815,6 +820,25 @@ def process_start_cycle_point(self) -> None:
             # Start from the initial point.
             self.start_point = self.initial_point
 
+    def cycle_point_warning(self, label1, order, label2):
+        """Centralized logic for warning that start, stop, initial and
+        final cycle points are not sensibly ordered.
+        """
+        point1 = getattr(self, label1 + '_point')
+        point2 = getattr(self, label2 + '_point')
+        if (
+            order == 'before' and point1 < point2
+            or order == 'after' and point1 > point2
+        ):
+            msg = (
+                f"{label1} cycle point '{point1}' will have no effect as"
+                f" it is {order} the {label2} cycle point '{point2}'."
+            )
+            # Capitalize() makes the ISO strings look very odd:
```

Comment:
Sorry, what does this mean?

Reply:
I don't see any need to capitalize the message in the first place

Reply:
> >>> 'foo bar baz 20250101T0000Z'.capitalize()
> 'Foo bar baz 20250101t0000z'

Not the point of your example, by since you've done it: task name case should not be messed with like this because task names are case sensitive: `Foo` could be an entirely different task from `foo`.

I guess the log message under discussion here starts with e.g. *initial cycle point*. Capitalizing that is less consequential (than task names), but I still don't think it should be done because `initial cycle point` is a case sensitive config item (I know the log message could be taken as referring to the concept, not the literal config item, but there's still some potential to confuse the user there).